crates/telio-starcast/src/starcast_peer.rs

```diff
@@ -0,0 +1,398 @@
+        let socket = UdpSocket::bind(SocketAddr::from(([127, 0, 0, 1], 0))).await?;
+        let wg_timer_tick_interval = interval(TIMER_UPDATE_PERIOD);
+        let secret_key = SecretKey::gen();
+
+        Ok(StarcastPeer {
+            task: Task::start(State {
+                secret_key,
+                channel,
+                socket,
+                tunnel: None,
+                wg_sock_addr: None,
+                wg_timer_tick_interval,
+                packets_present_in_tunnel: false,
+                sending_buffer: vec![0u8; MAX_PACKET],
+                receiving_buffer: vec![0u8; MAX_PACKET],
+                allowed_ips,
+            }),
+        })
+    }
+
+    // TODO figure out the implications and possible side effects of reconfiguring the starcast peer mid operation.
+    /// Configure (or reconfigure) the starcast peer.
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - A struct holding the configuration of the starcast peer.
+    ///
+    /// # Returns
+    ///
+    /// A result indicating wether the configuration was successful.
+    pub async fn configure(&self, config: Config) -> Result<(), Error> {
+        task_exec!(&self.task, async move |state| {
+            Ok(state.configure(config).await)
+        })
+        .await??;
+
+        Ok(())
+    }
+
+    /// Get the starcast virtual peer which can be treated as a regular peer by telio.
+    ///
+    /// # Returns
+    ///
+    /// A starcast peer.
+    pub async fn get_peer(&self) -> Result<Peer, Error> {
+        task_exec!(&self.task, async move |state| Ok(state.get_peer())).await?
+    }
+
+    /// Stop the starcast virtual peer.
+    pub async fn stop(self) {
+        let _ = self.task.stop().await.resume_unwind();
+    }
+}
+
+/// State for the starcast peer runtime where all the "working" members are stored and operated on.
+struct State {
+    /// Secret key of the starcast virtual peer.
+    secret_key: SecretKey,
+    /// Allowed IPs of the starcast virtual peer.
+    allowed_ips: Vec<IpNetwork>,
+    /// Interval at which the tunnel timers will be updated.
+    wg_timer_tick_interval: Interval,
+    /// Channel for sending decapsulated packets to the multicaster and receiving packets for encapsulation from it.
+    channel: Chan<Vec<u8>>,
+    /// Local socket for sending and receiving packets from WireGuard.
+    socket: UdpSocket,
+    /// WireGuard socket address to communicate encapsulated packets to and from Libtelio.
+    wg_sock_addr: Option<SocketAddr>,
+    /// WireGuard encryption tunnel for encapsulation/decapsulation of packets.
+    tunnel: Option<Tunn>,
+    /// Flag indicating that there are packets in the encryption tunnel internal buffer that need to be processed
+    /// before new packets can be received from the socket and decapsulated.
+    packets_present_in_tunnel: bool,
+    /// Buffer for outbound packets.
+    sending_buffer: Vec<u8>,
+    /// Buffer for inbound packets.
+    receiving_buffer: Vec<u8>,
```

Comment:
If we want to have buffers of static size, if would be safer to use `Box<[u8;2048]>`.